src/java/org/apache/cassandra/io/sstable/format/trieindex/PartitionIndex.java

```diff
@@ -78,6 +78,8 @@ public class PartitionIndex implements Closeable
     public static final long NOT_FOUND = Long.MIN_VALUE;
     public static final int FOOTER_LENGTH = 3 * 8;
 
+    private final ThreadLocal<Reader> cachedReaders = ThreadLocal.withInitial(() -> new Reader(this));
```

Comment:
I think we should (micro)benchmark that in isolation first; AFAIR I had similar question to @blambov some time ago and the conclusion was that the Reader is actually quite lightweight, most of the resources are loaded and being hold in `PartitionIndex` which is once per `SSTableReader` instance.

Reply:
it's possible that the cache key fix makes this obsolete since that was about 2/3 of the openReader
![Screenshot from 2023-09-14 17-21-55](https://github.com/datastax/cassandra/assets/42158/91c04404-6d50-4161-975a-2d24ec48c6ed)

Reply:
great, awesome catch :)

Reply:
I'll port that to oss